x-pack/plugins/cases/public/components/user_actions/index.tsx

```diff
@@ -0,0 +1,269 @@
+`;
+
+export const UserActions = React.memo(
+  ({
+    caseServices,
+    caseUserActions,
+    data: caseData,
+    fetchUserActions,
+    getRuleDetailsHref,
+    actionsNavigation,
+    isLoadingDescription,
+    isLoadingUserActions,
+    onRuleDetailsClick,
+    onShowAlertDetails,
+    onUpdateField,
+    renderInvestigateInTimelineActionComponent,
+    statusActionButton,
+    updateCase,
+    useFetchAlertData,
+    userCanCrud,
+  }: UserActionTreeProps) => {
+    const { detailName: caseId, subCaseId, commentId } = useCaseViewParams();
+    const [initLoading, setInitLoading] = useState(true);
+    const currentUser = useCurrentUser();
+
+    const [loadingAlertData, manualAlertsData] = useFetchAlertData(
+      getManualAlertIdsWithNoRuleId(caseData.comments)
+    );
+
+    const {
+      loadingCommentIds,
+      commentRefs,
+      selectedOutlineCommentId,
+      manageMarkdownEditIds,
+      handleManageMarkdownEditId,
+      handleOutlineComment,
+      handleSaveComment,
+      handleManageQuote,
+      handleUpdate,
+    } = useUserActionsHandler({ fetchUserActions, updateCase });
+
+    const MarkdownNewComment = useMemo(
+      () => (
+        <AddComment
+          id={NEW_COMMENT_ID}
+          caseId={caseId}
+          userCanCrud={userCanCrud}
+          ref={(element) => (commentRefs.current[NEW_COMMENT_ID] = element)}
+          onCommentPosted={handleUpdate}
+          onCommentSaving={handleManageMarkdownEditId.bind(null, NEW_COMMENT_ID)}
+          showLoading={false}
+          statusActionButton={statusActionButton}
+          subCaseId={subCaseId}
+        />
+      ),
+      [
+        caseId,
+        userCanCrud,
+        handleUpdate,
+        handleManageMarkdownEditId,
+        statusActionButton,
+        subCaseId,
+        commentRefs,
+      ]
+    );
+
+    useEffect(() => {
+      if (initLoading && !isLoadingUserActions && loadingCommentIds.length === 0) {
+        setInitLoading(false);
+        if (commentId != null) {
+          handleOutlineComment(commentId);
+        }
+      }
+    }, [commentId, initLoading, isLoadingUserActions, loadingCommentIds, handleOutlineComment]);
+
+    const descriptionCommentListObj: EuiCommentProps = useMemo(
+      () =>
+        getDescriptionUserAction({
+          caseData,
+          commentRefs,
+          manageMarkdownEditIds,
+          isLoadingDescription,
+          userCanCrud,
+          onUpdateField,
+          handleManageMarkdownEditId,
+          handleManageQuote,
+        }),
+      [
+        caseData,
+        commentRefs,
+        manageMarkdownEditIds,
+        isLoadingDescription,
+        userCanCrud,
+        onUpdateField,
+        handleManageMarkdownEditId,
+        handleManageQuote,
+      ]
+    );
+
+    const userActions: EuiCommentProps[] = useMemo(
+      () =>
+        caseUserActions.reduce<EuiCommentProps[]>(
+          (comments, userAction, index) => {
+            if (isUserActionTypeSupported(userAction.type)) {
```

Comment:
nit: could we flip this to do an early return instead:

```
if (!isUserActionTypeSupported(...) {
  return comments;
}
```

That way we avoid the indentation.